eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Proposals.hs

```diff
@@ -1,132 +1,451 @@
+      ]
+
+pfPParamUpdateL :: Lens' (PForest f era) (f (GovPurposeId 'PParamUpdatePurpose era))
+pfPParamUpdateL = lens pfPParamUpdate $ \x y -> x {pfPParamUpdate = y}
+
+pfHardForkL :: Lens' (PForest f era) (f (GovPurposeId 'HardForkPurpose era))
+pfHardForkL = lens pfHardFork $ \x y -> x {pfHardFork = y}
+
+pfCommitteeL :: Lens' (PForest f era) (f (GovPurposeId 'CommitteePurpose era))
+pfCommitteeL = lens pfCommittee $ \x y -> x {pfCommittee = y}
+
+pfConstitutionL :: Lens' (PForest f era) (f (GovPurposeId 'ConstitutionPurpose era))
+pfConstitutionL = lens pfConstitution $ \x y -> x {pfConstitution = y}
+
+pfrPParamUpdateL :: Lens' (PForest PRoot era) (PRoot (GovPurposeId 'PParamUpdatePurpose era))
+pfrPParamUpdateL = lens pfPParamUpdate $ \x y -> x {pfPParamUpdate = y}
+
+pfrHardForkL :: Lens' (PForest PRoot era) (PRoot (GovPurposeId 'HardForkPurpose era))
+pfrHardForkL = lens pfHardFork $ \x y -> x {pfHardFork = y}
+
+pfrCommitteeL :: Lens' (PForest PRoot era) (PRoot (GovPurposeId 'CommitteePurpose era))
+pfrCommitteeL = lens pfCommittee $ \x y -> x {pfCommittee = y}
+
+pfrConstitutionL :: Lens' (PForest PRoot era) (PRoot (GovPurposeId 'ConstitutionPurpose era))
+pfrConstitutionL = lens pfConstitution $ \x y -> x {pfConstitution = y}
+
+pfhPParamUpdateL :: Lens' (PForest PHierarchy era) (PHierarchy (GovPurposeId 'PParamUpdatePurpose era))
+pfhPParamUpdateL = lens pfPParamUpdate $ \x y -> x {pfPParamUpdate = y}
+
+pfhHardForkL :: Lens' (PForest PHierarchy era) (PHierarchy (GovPurposeId 'HardForkPurpose era))
+pfhHardForkL = lens pfHardFork $ \x y -> x {pfHardFork = y}
 
-instance EraPParams era => ToJSON (Proposals era)
+pfhCommitteeL :: Lens' (PForest PHierarchy era) (PHierarchy (GovPurposeId 'CommitteePurpose era))
+pfhCommitteeL = lens pfCommittee $ \x y -> x {pfCommittee = y}
 
-instance EraPParams era => NFData (Proposals era)
+pfhConstitutionL :: Lens' (PForest PHierarchy era) (PHierarchy (GovPurposeId 'ConstitutionPurpose era))
+pfhConstitutionL = lens pfConstitution $ \x y -> x {pfConstitution = y}
 
-instance EraPParams era => NoThunks (Proposals era)
+deriving instance Era era => Eq (PForest PRoot era)
+deriving instance Era era => Ord (PForest PRoot era)
+deriving instance Era era => Generic (PForest PRoot era)
+deriving instance Era era => NoThunks (PForest PRoot era)
+deriving instance Era era => NFData (PForest PRoot era)
+deriving instance Era era => Default (PForest PRoot era)
 
-instance Default (Proposals era) where
-  def = Proposals def
+deriving instance Era era => Eq (PForest PHierarchy era)
+deriving instance Era era => Ord (PForest PHierarchy era)
+deriving instance Era era => Generic (PForest PHierarchy era)
+deriving instance Era era => NoThunks (PForest PHierarchy era)
+deriving instance Era era => NFData (PForest PHierarchy era)
+deriving instance Era era => Default (PForest PHierarchy era)
+
+data Proposals era = Proposals
+  { pOMap :: !(OMap.OMap (GovActionId (EraCrypto era)) (GovActionState era))
```

Comment:
Not sure about this name, it expresses what's not so necessary (the type) and nothing about its meaning.  Should we go back to the `govActionsState` that we had at some point before?

Reply:
100%
Naming fields with types is an antipattern

Reply:
The new name now is `pProps`. Unless you guys think `pActionStates` is a better alternative.